addons/portal/controllers/mail.py

```diff
@@ -66,7 +66,7 @@ def _message_post_helper(res_model, res_id, message, token='', _hash=False, pid=
     author_id = request.env.user.partner_id.id if request.env.user.partner_id else False
 
     # Token Case: author is document customer (if not logged) or itself even if user has not the access
-    if token:
+    if token and not pid:
```

Comment:
Shoudn't we do the same fix (aka: check hash and pid, then token) ?

Reply:
You are right, I meant to change it then forgot, thanks :slightly_smiling_face: